modules/gke-cluster-standard/main.tf

```diff
@@ -128,7 +129,7 @@ resource "google_container_cluster" "cluster" {
       dynamic "auto_provisioning_defaults" {
         for_each = var.cluster_autoscaling.auto_provisioning_defaults != null ? [""] : []
         content {
-          boot_disk_kms_key = var.cluster_autoscaling.auto_provisioning_defaults.boot_disk_kms_key
+          boot_disk_kms_key = var.boot_disk_kms_key != null ? var.boot_disk_kms_key : var.cluster_autoscaling.auto_provisioning_defaults.boot_disk_kms_key
```

Comment:
why don't you use the existing field in auto provisioning defaults?

Reply:
As `boot_disk_kms_key` is a "root" variable, it makes sense to set the default KMS to the "root" variable if it's available.
I have changed the logic in my last commit to using in `auto_provisioning_defaults` first the variable ` var.cluster_autoscaling.auto_provisioning_defaults.boot_disk_kms_key` if it exists and fallback to `var.boot_disk_kms_key` if it exists.

What do you think?

Reply:
I don't think we need two variables for the same thing. :) And in general we always try to keep our variable spaces small. I am closing this, but thanks anyway.

Reply:
But without the setting into the `node_config` configuration, the constraint policy `constraints/gcp.restrictNonCmekServices` blocks everything and the module can't be used at all.

Do we have a solution?